### PR TITLE
[bugfix] Uppercase function 'undefind' for locale

### PR DESCRIPTION
**Pull request by Eliyas**
Handled undefined casses and warning messages.

### CODE REVIEW

**Comment by marwahaha** [comment]
@Eliyas - not sure what you are trying to do. Can you add some tests?

File: src/lib/create/local.js [hunk 1]
```diff
@@ -3,3 +3,7 @@ import { createLocalOrUTC } from './from-anything';
 export function createLocal (input, format, locale, strict) {
     return createLocalOrUTC(input, format, locale, strict, false);
 }
+
+export function createUTC (input, isUTC, format, locale, strict) {
```
Comment (marwahaha) [inline src/lib/create/local.js]: We already have this function here: https://github.com/moment/moment/blob/develop/src/lib/create/utc.js#L3

File: src/lib/locale/formats.js [hunk 2]
```diff
@@ -8,6 +8,10 @@ export var defaultLongDateFormat = {
 };
 
 export function longDateFormat (key) {
+    if (!key) {
```
Comment (marwahaha) [inline src/lib/locale/formats.js]: Why is this necessary? Who calls this method ?

File: src/lib/create/local.js (hunk 1 above)
Comment (Eliyas) [reply src/lib/create/local.js]: In the existing createLocal function we are hardcoded the param isUTC, so in compare.js isBefore(), isAfter() and isSame() methods need to pass isUTC while comparing UTC date.
https://github.com/moment/moment/blob/develop/src/lib/moment/compare.js

File: src/lib/locale/formats.js (hunk 2 above)
Comment (Eliyas) [reply src/lib/locale/formats.js]: I am trying to fix this issue. while calling longDateFormat without params. 
Uppercase function 'undefind' for locale. #4386

**Comment by Eliyas** [comment]
@marwahaha I have added test case for bugfix #4507. can you please have a look around

File: src/lib/locale/formats.js (hunk 2 above)
Comment (Eliyas) [reply src/lib/locale/formats.js]: isSameOrAfter() function in campare.js using isSame() and isAfter(), both the function need to pass isUtc boolean value to createLocal() function in order to check the utc string format.

**Comment by marwahaha** [comment]
Thanks for opening this request.
For now, we will not take this change.
